### PR TITLE
Feature swarm management

### PR DESCRIPTION
**Pull request by gerritvanos**
these frames will be used for comunication between SMM and swarm analytics

### CODE REVIEW

File: code/headers/frame_types.hpp [hunk 1]
```diff
@@ -396,6 +398,49 @@ namespace r2d2 {
...
+	
+    /*
+     * This frame will only be used with the python bus.
+     * The frame will be responsible for sending log data from 
+     * SMM to the swarm analytics module
+     *
+     * SMM wiki:
+     * https://github.com/R2D2-2019/R2D2-2019/wiki/Swarm-Management
+     */
+    R2D2_PACK_STRUCT
```
Comment (LRstudentHU) [inline code/headers/frame_types.hpp]: Please mark as a python struct

File: code/headers/frame_types.hpp [hunk 2]
```diff
@@ -396,6 +398,49 @@ namespace r2d2 {
...
+    struct frame_command_log_s {
+        // The current status of the command, for example received,processed,send etc.
+        // This is specified as an integer since swarm analytics will provide a table 
+       // containing the explanation for each status.
```
Comment (itzandroidtab) [inline code/headers/frame_types.hpp]: Indentation is wrong here

File: code/headers/frame_types.hpp [hunk 3]
```diff
@@ -396,6 +398,49 @@ namespace r2d2 {
...
+        uint16_t status;
+		
+        // This variable will contain the original recieved command type
+        char original_command;
+
+        // This variable will contain the original command data
+        char original_data;
+
```
Comment (LRstudentHU) [inline code/headers/frame_types.hpp]: Remove newline

File: code/headers/frame_types.hpp [hunk 4]
```diff
@@ -396,6 +398,49 @@ namespace r2d2 {
...
+	R2D2_PACK_STRUCT
+	struct frame_command_status_update_s {
+        // The command id for wich the status needs to be updated.
+        uint32_t cmd_id;
+		
+        // The current status of the command, for example received,processed,send etc.
+        // This is specified as an integer since swarm analytics will provide a table 
+        // containing the explanation for each status.
+        uint16_t status;
+
```
Comment (LRstudentHU) [inline code/headers/frame_types.hpp]: Extra newline

File: code/headers/frame_types.hpp [hunk 5]
```diff
@@ -396,6 +398,49 @@ namespace r2d2 {
...
+    };
+	
+    /*
+     * This frame will only be used with the python bus.
+     * The frame will be responsible for updating the status of a command.
+     *
+     * SMM wiki:
+     * https://github.com/R2D2-2019/R2D2-2019/wiki/Swarm-Management
+     */
+	R2D2_PACK_STRUCT
```
Comment (LRstudentHU) [inline code/headers/frame_types.hpp]: Please mark as a python struct

File: code/headers/frame_types.hpp (hunk 4 above)
Comment (itzandroidtab) [inline code/headers/frame_types.hpp]: Can you remove the newline here?

File: code/headers/frame_types.hpp [hunk 6]
```diff
@@ -406,4 +451,6 @@ namespace r2d2 {
     R2D2_INTERNAL_FRAME_HELPER(frame_path_step_s, PATH_STEP)
     R2D2_INTERNAL_FRAME_HELPER(frame_manual_control_s, MANUAL_CONTROL)    
     R2D2_INTERNAL_FRAME_HELPER(frame_movement_control_s, MOVEMENT_CONTROL)
+    R2D2_INTERNAL_FRAME_HELPER(frame_command_log_s, COMMAND_LOG)
+    R2D2_INTERNAL_FRAME_HELPER(frame_command_status_update_s, COMMAND_STATUS_UPDATE)
```
Comment (LRstudentHU) [inline code/headers/frame_types.hpp]: Can you please poison these structs? See the ui command for examples

**Review by LRstudentHU — CHANGES_REQUESTED** [review]

**Review by itzandroidtab — CHANGES_REQUESTED** [review]

File: code/headers/frame_types.hpp (hunk 6 above)
Comment (gerritvanos) [reply code/headers/frame_types.hpp]: i did not know this was needed

Comment (LRstudentHU) [reply code/headers/frame_types.hpp]: No worries, that was implemented last night and is not yet added to the readme.

**Review by LRstudentHU — APPROVED** [review]

**Review by itzandroidtab — APPROVED** [review]

File: code/headers/frame_types.hpp [hunk 7]
```diff
@@ -396,6 +398,47 @@ namespace r2d2 {
...
+     * This frame will only be used with the python bus.
+     * The frame will be responsible for sending log data from 
+     * SMM to the swarm analytics module
+     *
+     * SMM wiki:
+     * https://github.com/R2D2-2019/R2D2-2019/wiki/Swarm-Management
+     */
+    R2D2_PYTHON_FRAME
+    struct frame_command_log_s {
+        // The current status of the command, for example received,processed,send etc.
```
Comment (itzandroidtab) [inline code/headers/frame_types.hpp]: maybe some spaces after the ```,```

Comment (gerritvanos) [reply code/headers/frame_types.hpp]: could also put the example in []?

**Review by itzandroidtab — CHANGES_REQUESTED** [review]
see comment

File: code/headers/frame_types.hpp (hunk 7 above)
Comment (itzandroidtab) [reply code/headers/frame_types.hpp]: ?

File: code/headers/frame_types.hpp [hunk 8]
```diff
@@ -396,6 +398,47 @@ namespace r2d2 {
...
+     *
+     * SMM wiki:
+     * https://github.com/R2D2-2019/R2D2-2019/wiki/Swarm-Management
+     */
+    R2D2_PYTHON_FRAME
+    struct frame_command_status_update_s {
+        // The command id for wich the status needs to be updated.
+        uint32_t cmd_id;
+		
+        // The current status of the command, for example received, processed,send etc.
```
Comment (itzandroidtab) [inline code/headers/frame_types.hpp]: over here as well

**Review by itzandroidtab — APPROVED** [review]

**Pull request merged**